docs/proposals/20210222-kubelet-authentication.md

```diff
@@ -0,0 +1,319 @@
+---
+
+title: Cluster API Kubelet Authentication
+authors:
+  - "@randomvariable"
+  - "@yastij"
+reviewers:
+  - TBD
+creation-date: 2021-02-22
+last-updated: 2021-02-22
+status: implementable
+see-also:
+  - "/docs/proposals/20210222-cluster-api-node-bootstrapper.md"
+replaces:
+superseded-by:
+
+---
+
+# Cluster API Kubelet Authentication
+
+
+## Table of Contents
+
+A table of contents is helpful for quickly jumping to sections of a proposal and for highlighting
+any additional information provided beyond the standard proposal template.
+[Tools for generating](https://github.com/ekalinin/github-markdown-toc) a table of contents from markdown are available.
+
+- [Title](#title)
+  - [Table of Contents](#table-of-contents)
+  - [Glossary](#glossary)
+  - [Summary](#summary)
+  - [Motivation](#motivation)
+    - [Goals](#goals)
+    - [Non-Goals/Future Work](#non-goalsfuture-work)
+  - [Proposal](#proposal)
+    - [User Stories](#user-stories)
+      - [Story 1](#story-1)
+      - [Story 2](#story-2)
+    - [Requirements (Optional)](#requirements-optional)
+      - [Functional Requirements](#functional-requirements)
+        - [FR1](#fr1)
+        - [FR2](#fr2)
+      - [Non-Functional Requirements](#non-functional-requirements)
+        - [NFR1](#nfr1)
+        - [NFR2](#nfr2)
+    - [Implementation Details/Notes/Constraints](#implementation-detailsnotesconstraints)
+    - [Security Model](#security-model)
+    - [Risks and Mitigations](#risks-and-mitigations)
+  - [Alternatives](#alternatives)
+  - [Upgrade Strategy](#upgrade-strategy)
+  - [Additional Details](#additional-details)
+    - [Test Plan [optional]](#test-plan-optional)
+    - [Graduation Criteria [optional]](#graduation-criteria-optional)
+    - [Version Skew Strategy [optional]](#version-skew-strategy-optional)
+  - [Implementation History](#implementation-history)
+
+## Glossary
+
+
+- **OID:** Object Identifier defined by the International Telecommunications Union and used in PKI to identify attributes on certificates.
+
+- **PKI:** Public Key Infrastructure
+
+- **TPM:** Trusted Platform Module (TPM) is a specification defined by the Trusted Computing Group (TCG) that allows hosts to attest to their identity via PKI and a secure crypto-processor which may either be a separate chip, built into the CPU or a virtual device provided by the hypervisor.
+
+
+## Summary
+
+This proposal outlines a method to secure node registration within Cluster API, to solve 3 primary problems:
+- Solving a class of attacks involving node impersonation allowing an attacker to access secrets and volumes they shouldn’t by using hardware attestation of node identity.
+- Prevent token reuse in MachinePools where the token cannot be easily rotated.
```

Comment:
can you expand on this? I don't see it mentioned in the proposal?

Currently, the MachinePool bootstrap token does get rotated and has the same TTL (15mn) as Machine bootstrap tokens

Reply:
How does the bootstrap token get rotated? This would imply a change in userdata for the MachinePool. Is that supported on Azure?

Reply:
I've removed this text now that we've got to the bottom of what was happening in CAPA.